frontend/src/lib/modals/accounts/NewTransaction/TransactionForm.svelte

```
@@ -66,10 +62,10 @@
       return;
     }
     try {
-      const icp = convertNumberToICP(amount);
+      const tokens = TokenAmount.fromNumber({ amount, token }) as TokenAmount;
```

you need the cast? should we make `fromNumber` generic `.fromNumber<TokenAmount>`?

---

I'd like to make it return only `TokenAmount`. Generic not sure if it makes sense, it always returns that.

But now it uses `fromString` which doesn't always return a TokenAmount and might return someting else.

Should we maybe raise an error in `fromString` as well?

---

I am kind of open to any good solution that can spare a `as` cast 😉

If the case is just temporary, then maybe a `// TODO to be removed` I would say.

---

I will remove it from everywhere where I use `fromNumber`. There is no need for an extra comment here.

I also have the issue: https://dfinity.atlassian.net/browse/GIX-1005

But I didn't add a comment everywhere where this will change.

---

That do too, thanks